### PR TITLE
Closes #9556: Catch GeckoView Image RuntimeException

### LINKED ISSUE #9556: `java.lang.IllegalArgumentException` crash on nightly
Steps:
1. Go to www.reddit.com/r/videos
2. Click a video to expand it
3. Start playing the video
4. Click the small image of the video to close it

Expected behavior: closes the video
Actual behavior: Crash

Latest crash info (no associated crash report):
```
18edf71d-03ab-43a8-98a4-2d8cd2f52017
java.lang.IllegalArgumentException: Could not process image.
----
java.lang.IllegalArgumentException: Could not process image.
at org.mozilla.gecko.mozglue.GeckoLoader.nativeRun(Native Method)
at org.mozilla.gecko.GeckoThread.run(GeckoThread.java:101)
```

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by rocketsroger** [comment]
I'm open to ideas to how to test this.  I couldn't mock `Geckoview.Image`

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSessionDelegate.kt [hunk 1]
```diff
@@ -42,7 +43,8 @@ internal class GeckoMediaSessionDelegate(
                     withTimeoutOrNull(ARTWORK_RETRIEVE_TIMEOUT) {
                         it.getBitmap(ARTWORK_IMAGE_SIZE).await()
                     }
-                } catch (e: IllegalArgumentException) {
+                } catch (e: RuntimeException) {
+                    // TODO narrow this once GeckoView moved to ImageProcessingException
```
Comment (jonalmeida) [inline components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSessionDelegate.kt:47]: Keep a reference to the bug for this in the comment so we can come back to it and refer for later.

**Review by jonalmeida — APPROVED** [review]

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSessionDelegate.kt (hunk 1 above)
Comment (rocketsroger) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSessionDelegate.kt:47]: Agreed.  I'll ask for a Bugzilla issue and add it to the comments.  Thanks,

Comment (jonalmeida) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSessionDelegate.kt:47]: Feel free to file it yourself and let whoever know. I'm sure no one would mind. :)

Comment (rocketsroger) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSessionDelegate.kt]: Created the issue https://bugzilla.mozilla.org/show_bug.cgi?id=1691072.  Updated the dcomment as well.  Thanks,

Comment (rocketsroger) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/mediasession/GeckoMediaSessionDelegate.kt]: Updated to https://bugzilla.mozilla.org/show_bug.cgi?id=1689745

**Pull request merged**